docs/debug_boot.md

```diff
@@ -0,0 +1,55 @@
+# Debugging Your BlackParrot Boot
+
+## Nothing Happens!
+
+When integrating any IP into your project, the most common error is "nothing happens!". When dealing with a large, complex codebase, this is an intimidating blocker. This section details what to look for in your BlackParrot boot to smoke test the environment.
+
+If you're working on FPGA, first make sure BlackParrot can boot in RTL simulation.
+
+If you're working in RTL simulation, simply add these signals to the waveform environment. If they don't roughly match with this tutorial, double check that your environment emulates the testbench in bp\_top/test/tb/bp\_tethered/testbench.sv.
+
+If you're working on FPGA, _first make sure BlackParrot can boot in RTL simulation_. It's incredibly hard to debug FPGA environments, with the limited visibility that they provide. We (BlackParrot team) cannot provide debugging help for specific FPGA environments. Instead, we request all bug reports to be reproducible from within the tethered testbench provided. (If the bug cannot be reproduced by the specific construction of the tethered testbench, those reports are also welcome).
+
+## Boot Process (Method 1: NBF)
+
+The first method of boot involves a host tether which sets up BlackParrot for execution. The BlackParrot NBF boot happens in several steps. An NBF (network boot format) loader in the testbench takes in an NBF file, and converts it into a series of I/O writes into the core. This I/O fills the DRAM, cce microcode, and sets configuration registers within BlackParrot. Usage of nbf.py is shown below:
+
+
+    usage: nbf.py [-h] [--ncpus NCPUS] [--ucode ucode.mem] [--mem prog.mem]
+              [--config] [--checkpoint sample.nbf] [--skip_zeros]
+              [--addr_width ADDR_WIDTH]
+
+The default NPF boot sequence is as follows (waveforms shown from executing the "hello world" program):
+- reset is lowered. This must be lowered for 10 or more cycles. reset is raised.
+- The freeze register is set. This prevents BlackParrot from fetching or executing instructions.
+- I/O writes begin happening. There should be a single response for each request. These should look like the following:
+
+![I/O Writes](debug_io.png)
+
```

Comment:
The waveform diagrams are great, but can we provide example fully qualified paths? Would a DVE or other viewer configure file provide these?